### ISSUE TITLE
Questions about the official Tensorflow LR scheduler and train from scratch

### ISSUE THREAD

**Issue by akb46mayu** [zylo117/Yet-Another-EfficientDet-Pytorch#334]
Hi @zylo117,
Thanks for your great work! I looked into the official tensorflow configuration file as well as the training log, and found they may use 300 epochs (128 TPUs) with warmup+cos decay as the learning rate scheduler to train it from scratch.
•	If we use the same LR scheduler as google,  is that possible to train from scratch in a reasonable time frame (2weeks instead of a month) and reproduce similar results given 4-8 gpus? 
•	I was trying your model on a customized dataset and used all the training tricks mentioned in your repo (using pretrained weights & train from scratch), but found the model converges so slowly. In my previous experiments, 60-70 epochs is enough to train a light weight model like ssd from scratch and produce good mAPs, but when trying the effdet, under the same epochs the loss does not drop that much and the results are worse than the ssd model by 5-10 mAPs on my customized validation dataset.  Do you think it is just a learning rate issue (excluding the impact from data augmentation, anchor sizes, and bugs)? 
•	Could you please share your experience in tuning the learning rate with us briefly on how you train the model from scratch? According to my knowledge, did you tune the learning rate in a heuristic way instead of just one run? (like when the training loss decreases slowly, stop training and increase the LR; when the loss fluctuates, terminate the training and reduce LR; Using the best checkpoint achieved so far from validation to train the next round)

**Comment by zylo117** [comment]
1. I doubt it. I remember the paper says they used 32 TPU with 32G VRAM each, so 1024G VRAM in total. Correct me if I'm wrong. Without such VRAM, you can't manage to use such a large batchsize, so the BN will be the most affacted, especially when you train larger network like d3+. But training D0 shouldn't be a problem.
2. effdet is much harder to train compared to YOLO/SSD/RCNN, due to its super deep backbone and lots of activations. If it's not necessary, maybe try training with head_only first, it helps.
3. I didn't train it from scratch, but for your case, I think warmup+cos is a good choice. And train with adamw first, then switch to SGD.

**Comment by akb46mayu** [comment]
Hi @zylo117, thank you so much for your response.
1 Yes you are right. That was a typo.  Google used 128 batch size and 32 TPUs. Sorry for the mistake.
Batchsize and LR setting is really a big issue. I was using 64 as the batch size and found the initial learning rate did impact the training a lot. 
2 I agree. Train head_only first did help me get reasonable mAP within just 10-15 epochs.
3 I see. Thanks for the points, i am going to try warmup+cos decay later and think online augmentation should also make the model converge faster.
